Dockerfile

```diff
@@ -0,0 +1,30 @@
+# First stage: build environment
+FROM python:3.11-bullseye AS build-env
+
+# Set the working directory to /app
+WORKDIR /app
+
+# Copy the requirements file
+COPY requirements.txt .
+
+# Install the dependencies
+RUN pip install --no-cache-dir --upgrade pip && \
+    pip install --no-cache-dir -r requirements.txt
+
+# Second stage: runtime environment
+FROM python:3.11-slim
+
+# Set the working directory to /app
+WORKDIR /app
+
+RUN mkdir -p /root/.config/
+
+# Copy the script and libraries
+COPY ankerctl.py /app/ankerctl.py
+COPY libflagship /app/libflagship
+COPY cli /app/cli
+
+# Copy the installed dependencies from the build environment
+COPY --from=build-env /usr/local/lib/python3.11/site-packages /usr/local/lib/python3.11/site-packages
+
+ENTRYPOINT ["python", "/app/ankerctl.py"]
```

Comment:
Really nice dockerfile - clean separation between build-env and build product.

I'm not very familiar with packaging python in docker images, but would a venv maybe make sense? I suppose dealing with `/usr/local/lib/python3.11/site-packages` should pretty much always work, since we're explicitly depending on `python:3.11...`

Reply:
I'm not really a python expert and I haven't used venv almost at all. 

I checked the python3-slim container and I'm not seeing any virtual environment tools included. I suspect that docker is intending to be the virtual environment. 

```
docker run --entrypoint /bin/bash -it python:3.11-slim
root@3f8b068f8d04:/# venv
bash: venv: command not found
```